cmd/search/grep.go

```diff
@@ -21,23 +22,39 @@ import (
 )
 
 type Index struct {
-	Search     string
+	Search     []string
 	SearchType string
 	Context    int
 
 	MaxAge time.Duration
 }
 
 type Result struct {
+	// FailedAt is the time when the job failed.
 	FailedAt time.Time
+
+	// JobURI is the job detail page, e.g. https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-4.1/309
+	JobURI *url.URL
+
+	// FileType is the type of file where the match was found, e.g. "build-log" or "junit".
+	FileType string
+
+	// Trigger is "pull" or "build".
+	Trigger string
+
+	// Name is the name of the job, e.g. release-openshift-ocp-installer-e2e-aws-4.1 or pull-ci-openshift-origin-master-e2e-aws.
+	Name string
+
+	// Number is the job number, e.g. 309 for origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-4.1/309 or 5466 for origin-ci-test/pr-logs/pull/openshift_installer/1650/pull-ci-openshift-installer-master-e2e-aws/5466.
+	Number int
 }
 
 type ResultMetadata interface {
-	MetadataFor(path string) (Result, bool)
+	MetadataFor(path string) (*Result, bool)
 }
 
 type CommandGenerator interface {
-	Command(*Index) (cmd string, args []string)
+	Command(index *Index, search string) (cmd string, args []string)
```

Comment:
I don't understand this change.  Search is in index?

Reply:
> I don't understand this change. Search is in index?

This PR makes `Index.Search` a `[]string`, so the new argument here is "of the strings we're interested, *this* command should be for *this* string.  This could also be an index `int` if we wanted to go that way.